store.go

```diff
@@ -2788,6 +2788,9 @@ func (s *store) Layers() ([]Layer, error) {
 	if err != nil {
 		return nil, err
 	}
+	if err := lstore.LoadLocked(); err != nil {
+		return nil, err
+	}
```

Comment:
This unconditionally reloads the store contents using a write lock here, in addition to where it's conditionally done under a read lock on line 2805.  We'll probably want to drop `lstore` from the initial value of the slice that's constructed on line 2800 to avoid that extra bit of work.

Reply:
Alright I put another commit on top of my changes to see if the new logic breaks anything.